src/main/scala/workflow/graph/DefaultOptimizer.scala

```diff
@@ -0,0 +1,11 @@
+package workflow.graph
+
+/**
+ * The default Pipeline optimizer used when executing pipelines.
+ */
+object DefaultOptimizer extends Optimizer {
+  protected val batches: Seq[Batch] =
+    Batch("Load Saved State", Once, ExtractSaveablePrefixes, SavedStateLoadRule, DanglingNodeRemovalRule) ::
```

Comment:
We need the correct prefix mapping available post-optimization, at execution time. This is to save the result of a node in the global state w/ the correct prefix. This is why every rule needs to maintain the prefix mappings for the nodes whose results we will want to save.